src/main/java/iroha/validation/transactions/provider/impl/BasicTransactionProvider.java

```diff
@@ -34,67 +25,44 @@
 
   private static final Logger logger = LoggerFactory.getLogger(BasicTransactionProvider.class);
 
-  private final IrohaAPI irohaAPI;
-  // BRVS account id to query Iroha
-  private final String accountId;
-  // BRVS keypair to query Iroha
-  private final KeyPair keyPair;
   private final TransactionVerdictStorage transactionVerdictStorage;
   private final CacheProvider cacheProvider;
   private boolean isStarted;
   private final ScheduledExecutorService executorService = Executors.newScheduledThreadPool(2);
   // Accounts to monitor pending tx
   private final Set<String> accountsToMonitor = new HashSet<>();
 
+  private final IrohaHelper irohaHelper;
+
   @Autowired
-  public BasicTransactionProvider(IrohaAPI irohaAPI,
-      String accountId,
-      KeyPair keyPair,
+  public BasicTransactionProvider(
       TransactionVerdictStorage transactionVerdictStorage,
-      CacheProvider cacheProvider) {
-    Objects.requireNonNull(irohaAPI, "Iroha API must not be null");
-    if (Strings.isNullOrEmpty(accountId)) {
-      throw new IllegalArgumentException("Account ID must not be neither null or empty");
-    }
-    Objects.requireNonNull(keyPair, "Keypair must not be null");
+      CacheProvider cacheProvider,
+      IrohaHelper irohaHelper
+  ) {
     Objects.requireNonNull(transactionVerdictStorage, "TransactionVerdictStorage must not be null");
     Objects.requireNonNull(cacheProvider, "CacheProvider must not be null");
 
-    this.irohaAPI = irohaAPI;
-    this.accountId = accountId;
-    this.keyPair = keyPair;
     this.transactionVerdictStorage = transactionVerdictStorage;
     this.cacheProvider = cacheProvider;
+    this.irohaHelper = irohaHelper;
   }
 
-  /**
-   * {@inheritDoc}
-   */
-  @Override
-  public synchronized Observable<Transaction> getPendingTransactionsStreaming() {
+  public synchronized void start() {
```

Comment:
I do recommend to hide `start` from an external user to `getPendingTransactionsStreaming()` as it was before